### PR TITLE
98144 Sign-in widget: Update verify alert to use service-specific buttons and messaging

### PR DESCRIPTION
**Pull request by nichia**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
- Change the verify alert to use service-specific buttons that is relevant to the CSP user signed in with
- Update verify alert messaging 

## Related issue(s)

- department-of-veterans-affairs/va.gov-team/issues/98144
- department-of-veterans-affairs/va.gov-team/issues/97279

## Testing done

- Add/update unit tests
- Manual testing in local development:
  - 'yarn watch` from `vets-website`
  - `foreman start -m all=1,clamd=0,freshclam=0` from `vets-api`
  - Visit `http://localhost:3001/sign-in/mocked-auth` to use mocked authentication for Credential Service Provider:
    - ID.me  (profile: joe.niquette+idmetest5@oddball.io | MPI not mocked)
    - Login.gov (profile: joe.niquette+lgovial1@oddball.io | MPI not mocked)
    - no existing test users for My HealtheVet, so manual code change to view the SignInOtherAccount for MHV Basic account)
    - Visit `http://localhost:3001/health-care/schedule-view-va-appointments/` to check that relevant alert is rendered

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |    Screenshot 2024-12-12 at 10 11 43 AM    |   Screenshot 2024-12-12 at 10 05 58 AM    |
| Desktop  |    Screenshot 2024-12-12 at 10 13 38 AM   |   Screenshot 2024-12-10 at 9 02 01 PM   |
| Mobile  |   Screenshot 2024-12-12 at 10 11 43 AM     |    Screenshot 2024-12-12 at 10 05 26 AM   |
| Desktop |   Screenshot 2024-12-12 at 10 13 38 AM     |  Screenshot 2024-12-10 at 9 03 47 PM  |
| Mobile |   Screenshot 2024-12-12 at 10 11 43 AM      |   Screenshot 2024-12-16 at 2 05 18 PM  |
| Desktop |   Screenshot 2024-12-12 at 10 13 38 AM     |   Screenshot 2024-12-16 at 2 04 27 PM   |

## What areas of the site does it impact?

These changes impact _widgets_ that get embedded on static pages. The following are some paths that embed widgets that used this alert.

- `/health-care/schedule-view-va-appointments/`
- `/health-care/refill-track-prescriptions/`
- `/health-care/secure-messaging/`
- `/health-care/view-test-and-lab-results/`
- `health-care/get-medical-records/`

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/static-pages/cta-widget/components/messages/Verify.jsx [hunk 1]
```diff
@@ -1,36 +1,75 @@
...
+  signInService: CSP_IDS.ID_ME,
+};
+
 Verify.propTypes = {
-  primaryButtonHandler: PropTypes.func.isRequired,
-  serviceDescription: PropTypes.string.isRequired,
-  headerLevel: PropTypes.string,
+  headerLevel: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  serviceDescription: PropTypes.string,
+  signInService: PropTypes.oneOf([CSP_IDS.ID_ME, CSP_IDS.LOGIN_GOV]),
```
Comment (radavis) [inline src/applications/static-pages/cta-widget/components/messages/Verify.jsx:72]: I'm wondering if it is possible for this component to be called with other `CSP_IDS` and what the result would be.

Comment (radavis) [reply src/applications/static-pages/cta-widget/components/messages/Verify.jsx:72]: After reading the `getVerifyAlert` function in `cta-widget/index.js`, the above shouldn't be a concern.

**Review by radavis — DISMISSED** [review]
LGTM!

**Comment by dcloud** [comment]
I thought I could `yarn watch --env entry=mhv-landing-page,auth,static-pages` and `yarn mock-api --responses src/applications/mhv-landing-page/mocks/api` with a `USER_MOCKS.MHV_BASIC_ACCOUNT` to see the "You need to sign in with a different account" but am only able (due to local content-build issues) to look at `http://localhost:3001/health-care/schedule-view-va-appointments/`, which shows the "Go to My HealtheVet" alert (a separate change for a separate ticket).

I think the changes look good and make sense given you've edited the right places. Just saying testing with an MHV login is hard, and probably not super valuable since this work is in service of removing that login type

**Review by dcloud — DISMISSED** [review]
These changes look good to me.

**Comment by asg5704** [comment]
@nichia These changes look good to me. Ready to approve once this moves to Ready

**Review by asg5704 — DISMISSED** [review]

**Comment by fmccaf1** [comment]
Looks right @nichia, thanks!

**Review by dcloud — DISMISSED** [review]
Looks good to me

**Review by asg5704 — DISMISSED** [review]

**Review by asg5704 — APPROVED** [review]

**Review by dcloud — APPROVED** [review]

**Review by IGallupSoCo — APPROVED** [review]
LGTM

**Pull request merged**